src/rust/engine/src/externs.rs

```diff
@@ -555,6 +548,18 @@ impl TypeIdBuffer {
   pub fn to_vec(&self) -> Vec<TypeId> {
     with_vec(self.ids_ptr, self.ids_len as usize, |vec| vec.clone())
   }
+
```

Comment:
I might do something like:
```rust
pub trait FFIBuffer<T, E> {
  fn to_vec(&self) -> Vec<T>;
  fn unwrap_one(&self) -> Result<T, E>;
}

impl FFIBuffer<Value, String> for HandleBuffer { ... }

impl FFIBuffer<TypeId, String> for TypeIdBuffer { ... }
```

but there might be a better way to split that off e.g. requiring each type impl `*_ptr()` and `*_len()` methods, and then implementing `to_vec()` and `unwrap_one()` generically, but that might be too far for right now.

Reply:
I'm not sure the generics are worth it in this case... if we grow a third copy of this method we can dive in there =)

Reply:
Added a comment to this effect.